src/Apps/Legion/Adapters/Standalone/DataManagement/ADXAdapter.ts

```diff
@@ -0,0 +1,155 @@
+import axios from 'axios';
+import AdapterMethodSandbox from '../../../../../Models/Classes/AdapterMethodSandbox';
+import { ComponentErrorType } from '../../../../../Models/Constants/Enums';
+import { IAuthService } from '../../../../../Models/Constants/Interfaces';
+import BaseAdapter from '../../BaseAdapter';
+import { DataManagementAdapterData } from './Models/DataManagementAdapter.data';
+import {
+    IDataManagementAdapter,
+    IGetDataAdapterParams,
+    IPushDataAdapterParams
+} from './Models/DataManagementAdapter.types';
+
+export default class ADXAdapter
+    extends BaseAdapter
+    implements IDataManagementAdapter {
+    connectionSource: string;
+
+    constructor(authService: IAuthService, connectionSource: string) {
+        super(authService);
+        this.connectionSource = connectionSource;
+        this.authService.login();
+    }
+
+    /** fetches database list or table list based on params
+     * TODO: explore function overloading
+     */
+    async getData(params: IGetDataAdapterParams) {
```

Comment:
prolly two methods here IMO.  getData isnt very descriptive and its best if it doesnt fork the logic based on params IMO

Reply:
yeah, my intention was like creating this IDataManagementAdapter interface which all these adapters are implementig to encapsulate all possible data management services and keep it as generic as possible for the future. But okay, I am going to break it into more granular methods in the interface as well and change the implementation here too.